gadopt/time_stepper.py

```diff
@@ -67,7 +67,7 @@ def __init__(
         self.test = firedrake.TestFunction(solution.function_space())
         self.solution = solution
         self.fields = fields
-        self.dt = dt
+        self.dt = float(dt)
```

Comment:
Why was this necessary? How haven't we hit an issue with this before?

Reply:
It is not necessary. However, it seemed reasonable to me to ensure that `self.dt` is a `float` if we ensure that `self.dt_const` is a `Constant`.